notebook/notebookapp.py

```diff
@@ -2294,16 +2325,36 @@ def list_running_servers(runtime_dir=None):
             with io.open(os.path.join(runtime_dir, file_name), encoding='utf-8') as f:
                 info = json.load(f)
 
-            # Simple check whether that process is really still running
+            # Actively check whether that process is really available via an HTTP request
             # Also remove leftover files from IPython 2.x without a pid field
-            if ('pid' in info) and check_pid(info['pid']):
-                yield info
-            else:
-                # If the process has died, try to delete its info file
+            response = kernel_request(info, path=url_path_join(info.get('base_url') or '/','/api'))
+            def flush_info_file():
                 try:
                     os.unlink(os.path.join(runtime_dir, file_name))
                 except OSError:
-                    pass  # TODO: This should warn or log or something
+                    pass
+
+            version_dict = {}
+            try:
+                version_dict.update(json.loads(response.body.decode()))
+            except:
+                flush_info_file()
+            else:
+                received_version = version_dict.get('version', '0.0.0')
+                msg_special = "Jupyter Notebook HIGHER VERSION detected: %s, current:%s" %(
+                    received_version, __version__)
+                msg_upgrade = "Jupyter Notebook Upgrade REQUIRED: >=%s, got:%s(info file removed)" % (
+                    __version__, received_version)
+
+                if JUPYTER_NOTEBOOK_TAG != version_dict.get('module'):
+                    raise MultipleInstanceError(msg_upgrade)
+                else:
+                    if __version__ < received_version:
```

Comment:
Let's just check if the versions are different and issue a warning message.  I think its much easier to let the user know that there are multiple versions in use and let them decide which version is what they want.
```suggestion
                    if __version__ != received_version:
```

Reply:
we MUST conduct users by make policies:
1. we encourage users to keep update with latest version of Jupyter Notebook;
```
     if JUPYTER_NOTEBOOK_TAG != version_dict.get('module'):
         raise MultipleInstanceError(msg_upgrade)
```

2. we encourage users to hack our design and mark with greater version number than official latest version;
```
    if __version__ < received_version:
        print(msg_special)
        #hacked by users ;-)
```

3. forward compatibility(if possible):
```
        __version__ >= received_version and JUPYTER_NOTEBOOK_TAG == version_dict.get('module')
    #users can run multi-version of Jupyter Notebook if it marked with JUPYTER_NOTEBOOK_TAG
```